### DISCUSSION TITLE
Unable to get listings using client.get_item_listings()

### DISCUSSION THREAD

**Discussion by pudjojotaro** [somespecialone/aiosteampy#68] [Q&A] [answered]
```
import asyncio
import json
from pathlib import Path
from fake_useragent import UserAgent

# Cookie file to store session cookies
cookie_file = Path("cookie.json")

async def main():
    from aiosteampy import SteamClient, Currency, AppContext
    from aiosteampy.helpers import restore_from_cookies
    from aiosteampy.utils import get_jsonable_cookies

    # Get a random user agent
    random_user_agent = get_random_user_agent()

    # Create SteamClient instance with credentials and user agent
    client = SteamClient(
    )

    # Restore session from cookies if the cookie file exists
    if cookie_file.is_file():
        with cookie_file.open("r") as f:
            cookies = json.load(f)
        await restore_from_cookies(cookies, client)
    else:
        # Perform a login and save cookies
        await client.login()

    try:
        listings, total_count = await client.get_item_listings("Dreams & Nightmares Case", 730)
        print(listings, total_count)

    except Exception as e:
        print(f"Error during Steam client operations: {e}")

    finally:
        # Save cookies to the file
        with cookie_file.open("w") as f:
            json.dump(get_jsonable_cookies(client.session), f)

        # Ensure the client session is properly closed
        await client.session.close()

if __name__ == "__main__":
    import platform

    # Set Windows-specific event loop policy
    platform.system() == "Windows" and asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

    asyncio.run(main())
```

I was testing the library using the code examples that I found in the "examples" folder as well as reading the documentation on [https://aiosteampy.somespecial.one/public/](url) . After combining the two together, I managed to complete the login and some of the methods (like getting histogram) worked fine, however I never managed to run the client.get_item_listings() as it always exited with the following error message:

`Error during Steam client operations: 'int' object has no attribute 'value'`

Am I doing something wrong? Is there a different way the listings should be retrieved? Thank you in advance.

**Comment by somespecialone** [comment, marked as answer]
Hi! **Firstly**, that is a right way to get item listings:
```python
listings, count, last_modified = await client.get_item_listings("Dreams & Nightmares Case",  App.CS2)
```
Second argument is an `App` enum, not just `int`.

**Secondly**, `Dreams & Nightmares Case` has no listings on market, as it is realised through the order table.

And finally, **third point**: please, pay attention, that documentation after [enums](https://aiosteampy.somespecial.one/enums/) page is out of date and corresponds to old version.

**Reply by pudjojotaro** [reply]
Thank you for your response, I will try it out today and let you know how it goes!
